src/plugins/crypto/README.md

```diff
@@ -109,62 +110,62 @@ Set the CMake variables `OPENSSL_INCLUDE_DIR` and `OPENSSL_LIBRARIES` to your de
 
 The crypto plugin will encrypt and decrypt values using AES-256 in CBC mode.
 
-The key derivation is still WIP.
+The GPG binary is required for handling the cryptographic keys.
 
-### Planned Features ###
+At the moment the plugins will only run on UNIX/Linux-like systems, that implement `fork ()`.
 
-- Encryption of values
-- Decryption of values
-- Key derivation by metadata (password provided within a meta-key)
-- Key derivation by using a specified key-file (like the SSH client does)
-- Key derivation by utilizing the pgp-agent
+## Examples ##
+
+To mount a backend with the gcrypt plugin variant that uses the GPG key 9CCC3B514E196C6308CCD230666260C14A525406, use:
 
-The encryption and decryption of values is a straight forward process, once the key and IV are supplied.
-The key-derivation process in a library is a bit tricky.
+	kdb mount test.ecf user/t crypto_gcrypt "crypto/key=9CCC3B514E196C6308CCD230666260C14A525406"
 
-The crypto plugin itself can hold configuration data about the order of the applied key derivation functions.
-For example, if a password is provided by a meta-key, it will be used, otherwise we look for a key file.
-If no key file has been configured, we try to trigger the PGP agent, etc.
+Now you can specify a key `user/t/a` and protect its content by using:
 
-The following example configuration illustrates this concept:
+	kdb set user/t/a
+	kdb setmeta user/t/a crypto/encrypt 1
+	kdb set user/t/a "secret"
```

Comment:
I managed to reproduce the error you posted before. And I kind of fixed it by excluding the "spec" namespace from encryption and decryption. But setting the metakey "crypto/encrypt" in spec namespace does not trigger encryption (I think encryption should start?).

	root@258e61a6f05c:~/libelektra/build# bin/kdb mount test.ecf user/t crypto_gcrypt /gpg/key=$KEY
	root@258e61a6f05c:~/libelektra/build# bin/kdb global-mount
	root@258e61a6f05c:~/libelektra/build# bin/kdb setmeta spec/t/a crypto/encrypt 1
	root@258e61a6f05c:~/libelektra/build# bin/kdb set user/t/a "secret"
	Create a new key user/t/a with string secret
	root@258e61a6f05c:~/libelektra/build# bin/kdb get user/t/a
	secret
	root@258e61a6f05c:~/libelektra/build# xxd ~/.config/test.ecf 
	0000000: 6b64 624f 7065 6e20 310a 6b73 4e65 7720  kdbOpen 1.ksNew 
	0000010: 310a 6b65 794e 6577 2039 2037 0a75 7365  1.keyNew 9 7.use
	0000020: 722f 742f 6100 7365 6372 6574 000a 6b65  r/t/a.secret..ke
	0000030: 7945 6e64 0a6b 7345 6e64 0a              yEnd.ksEnd.
	root@258e61a6f05c:~/libelektra/build# bin/kdb setmeta spec/t/a crypto/encrypt 1
	root@258e61a6f05c:~/libelektra/build# xxd ~/.config/test.ecf 
	0000000: 6b64 624f 7065 6e20 310a 6b73 4e65 7720  kdbOpen 1.ksNew 
	0000010: 310a 6b65 794e 6577 2039 2037 0a75 7365  1.keyNew 9 7.use
	0000020: 722f 742f 6100 7365 6372 6574 000a 6b65  r/t/a.secret..ke
	0000030: 7945 6e64 0a6b 7345 6e64 0a              yEnd.ksEnd.

I still have to set the meta-key in "user" namespace.

	root@258e61a6f05c:~/libelektra/build# bin/kdb setmeta user/t/a crypto/encrypt 1
	root@258e61a6f05c:~/libelektra/build# xxd ~/.config/test.ecf 
	0000000: 6b64 624f 7065 6e20 310a 6b73 4e65 7720  kdbOpen 1.ksNew 
	0000010: 310a 6b65 794e 6577 2039 2033 320a 7573  1.keyNew 9 32.us
	0000020: 6572 2f74 2f61 0033 1b14 838e 58d1 ea91  er/t/a.3....X...
	0000030: 27eb f401 869f fac4 8b9e 04e8 0152 67c6  '............Rg.
	0000040: da96 6b4d 6426 c30a 6b65 794d 6574 6120  ..kMd&..keyMeta 
	0000050: 3720 310a 6269 6e61 7279 0000 0a6b 6579  7 1.binary...key
	0000060: 4d65 7461 2031 3520 320a 6372 7970 746f  Meta 15 2.crypto
	0000070: 2f65 6e63 7279 7074 0031 000a 6b65 794d  /encrypt.1..keyM
	0000080: 6574 6120 3132 2031 370a 6372 7970 746f  eta 12 17.crypto
	0000090: 2f73 616c 7400 2044 2d4e 333e 2524 4f4e  /salt. D-N3>%$ON
	00000a0: 2a50 4171 3c79 000a 6b65 7945 6e64 0a6b  *PAq<y..keyEnd.k
	00000b0: 7345 6e64 0a                             sEnd.